bug: range型dimension閾値(7-9)がgoal addでパースエラー

## Summary
`goal add` で dimension の threshold_type が range の場合（例: `sleep_duration_hours:range:7-9`）、CLIが `invalid value in --dim` エラーで失敗する。

## Root Cause
TBD — dimension推論でLLMが `7-9` のようなrange値を返すが、`--dim` パーサーがrange形式を受け付けない可能性。

## Reproduction
```bash
node dist/cli-runner.js goal add --title "健康的な生活習慣を身につける" --yes
```

出力:
```
--- Inferred dimensions ---
  2. sleep_duration_hours  [range]  threshold: 7-9
--- Auto-accepted (--yes) ---
[ERROR] Error: invalid value in --dim "sleep_duration_hours:range:7-9". Check type/value combination.
```

## Expected Behavior
range型dimensionが正常に登録される。

## Related
Dogfooding round 3, dimension auto-inference (#304)

--- a/src/cli/commands/goal-infer.ts
+++ b/src/cli/commands/goal-infer.ts
@@ -31,7 +31,7 @@ function buildInferPrompt(title: string): string {
 Threshold types:
 - min: value must be >= X (e.g. test coverage >= 80%)
 - max: value must be <= X (e.g. bug count <= 5)
-- range: value must be between low-high (e.g. "10-20")
+- range: value must be between low,high (e.g. "10,20")
 - present: boolean check — something must exist (value: "true" or "false")
 - match: exact match required (e.g. status must equal "published")
 

--- a/src/cli/commands/goal-utils.ts
+++ b/src/cli/commands/goal-utils.ts
@@ -92,7 +92,9 @@ export function buildThreshold(spec: RawDimensionSpec): Threshold | null {
 
   if (spec.type === "range") {
     if (!spec.value) return null;
-    const [lowStr, highStr] = spec.value.split(",");
+    let parts = spec.value.split(",");
+    if (parts.length < 2) parts = spec.value.split("-");
+    const [lowStr, highStr] = parts;
     const low = parseFloat(lowStr ?? "");
     const high = parseFloat(highStr ?? "");
     if (isNaN(low) || isNaN(high)) return null;

--- a/tests/cli/goal-utils.test.ts
+++ b/tests/cli/goal-utils.test.ts
@@ -1,0 +1,38 @@
+import { describe, it, expect } from "vitest";
+import { buildThreshold } from "../../src/cli/commands/goal-utils.js";
+
+describe("buildThreshold", () => {
+  describe("range type", () => {
+    it("parses comma-separated range", () => {
+      expect(buildThreshold({ name: "x", type: "range", value: "7,9" })).toEqual({
+        type: "range",
+        low: 7,
+        high: 9,
+      });
+    });
+
+    it("parses hyphen-separated range", () => {
+      expect(buildThreshold({ name: "x", type: "range", value: "7-9" })).toEqual({
+        type: "range",
+        low: 7,
+        high: 9,
+      });
+    });
+
+    it("parses decimal comma-separated range", () => {
+      expect(buildThreshold({ name: "x", type: "range", value: "10.5,20.5" })).toEqual({
+        type: "range",
+        low: 10.5,
+        high: 20.5,
+      });
+    });
+
+    it("returns null when value is missing", () => {
+      expect(buildThreshold({ name: "x", type: "range", value: undefined })).toBeNull();
+    });
+
+    it("returns null when value is not parseable", () => {
+      expect(buildThreshold({ name: "x", type: "range", value: "abc" })).toBeNull();
+    });
+  });
+});

